cmd/storagenode/main.go

```diff
@@ -291,6 +303,49 @@ func cmdDiag(cmd *cobra.Command, args []string) (err error) {
 	return nil
 }
 
+// maps deprecated config values to new values if applicable
+func mapConfigs(log *zap.Logger) {
+	type config struct {
+		new     *string
+		newFlag string
+		old     *string
+		oldFlag string
+	}
+	configs := []config{
+		{
+			new:     &runCfg.Contact.ExternalAddress,
+			newFlag: "contact.external-address",
+			old:     &runCfg.Kademlia.ExternalAddress,
+			oldFlag: "kademlia.external-address",
+		},
+		{
+			new:     &runCfg.Operator.Wallet,
+			newFlag: "operator.wallet",
+			old:     &runCfg.Kademlia.Operator.Wallet,
+			oldFlag: "kademlia.operator.wallet",
+		},
+		{
+			new:     &runCfg.Operator.Email,
+			newFlag: "operator.email",
+			old:     &runCfg.Kademlia.Operator.Email,
+			oldFlag: "kademlia.operator.email",
+		},
+	}
+
+	for _, config := range configs {
+		if *config.new != "" && *config.old != "" {
+			log.Warn("Both " + config.newFlag + " and " + config.oldFlag + " are designated in your config.yaml. " +
```

Comment:
nit (feel free to ignore) - this might look better using zap arguments or with a formatted string. Same applies to other log lines.